### PR TITLE
`azurerm_role_assignment` - fix id parsing for root or provider scoped role assignments

### PR DESCRIPTION
**Pull request by Felix-Franz**
## Community Note

* Please vote on this PR by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original PR to help the community and maintainers prioritize for review
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for PR followers and do not help prioritize for review

## Description

With the azurerm_role_assignment resource it is possible to make RBAC assignments on root or resource provider scope.
The functionality was added in #26663.

The creation succeeds, but the provider cannot parse the resource ID from the ARM-response.
Following error is displayed:
```
Error: could not parse Role Assignment ID "/providers/Microsoft.Authorization/roleAssignments/23456781-2349-8764-5631-234567890121"
```

The PR is intended to fix the parsing of the resource ID and thus enable an assignment on root or resource provider scope.

## PR Checklist

- [x] I have followed the guidelines in our [Contributing Documentation](../blob/main/contributing/README.md).
- [x] I have checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change.
- [x] I have checked if my changes close any open issues. If so please include appropriate [closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) below.
- [ ] I have updated/added Documentation as required written in a helpful and kind way to assist users that may be unfamiliar with the resource / data source.
- [x] I have used a meaningful PR title to help maintainers and other users understand this change and help prevent duplicate work. 
For example: “`resource_name_here` - description of change e.g. adding property `new_property_name_here`”

## Changes to existing Resource / Data Source

- [x] I have added an explanation of what my changes do and why I'd like you to include them (This may be covered by linking to an issue above, but may benefit from additional explanation).
- [x] I have written new tests for my resource or datasource changes & updated any relevent documentation.
- [x] I have successfully run tests with my changes locally. If not, please provide details on testing challenges that prevented you running the tests.

## Testing 

- [ ] My submission includes Test coverage as described in the [Contribution Guide](../blob/main/contributing/topics/guide-new-resource.md) and the tests pass. (if this is not possible for any reason, please include details of why you did or could not add test coverage)

## Change Log

Below please provide what should go into the changelog (if anything) conforming to the [Changelog Format documented here](../blob/main/contributing/topics/maintainer-changelog.md).

* `azurerm_resource` - support for the `thing1` property [GH-00000]

This is a (please select all that apply):

- [x] Bug Fix
- [ ] New Feature (ie adding a service, resource, or data source)
- [ ] Enhancement
- [ ] Breaking Change

## Related Issue(s)
Fixes #0000

> [!NOTE] 
> If this PR changes meaningfully during the course of review please update the title and description as required.

### CODE REVIEW

**Comment by Felix-Franz** [comment]
@stephybun @catriona-m I am taging you because you were involved in the original PR.
Could you please review this PR?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this PR @Felix-Franz.

I'm wondering whether the [parser used in the delete call](https://github.com/hashicorp/terraform-provider-azurerm/blob/fa1343303ae58c0854f7293f62db1108bea7ea77/internal/services/authorization/role_assignment_resource.go#L409-L429) might also require an update? Looking at what it's doing I believe it will be fine, but since our testing credentials lack the privileges to create role assignments on these scopes, it'd help us out if you could confirm whether with this fix you can also delete the role assignments without issue?

File: internal/services/authorization/parse/role_assignment.go [hunk 1]
```diff
@@ -179,6 +189,22 @@ func RoleAssignmentID(input string) (*RoleAssignmentId, error) {
 		}
 		roleAssignmentId.Name = idParts[1]
 		roleAssignmentId.ManagementGroup = strings.TrimPrefix(idParts[0], "/providers/Microsoft.Management/managementGroups/")
+	case strings.HasPrefix(input, "/providers/") && !strings.HasPrefix(input, "/providers/Microsoft.Authorization/roleAssignments"):
+		idParts := strings.Split(input, "/providers/Microsoft.Authorization/roleAssignments/")
+		if len(idParts) != 2 {
+			return nil, fmt.Errorf("could not parse Role Assignment ID %q for Ressource Provider", input)
```
[the remark this answers is not in the document]
Comment (Felix-Franz) [reply internal/services/authorization/parse/role_assignment.go:195]: Hello @stephybun,

I have tested the delete again locally (see [destroy.pdf](https://github.com/user-attachments/files/17201149/destroy.pdf)).
As expected, the parser from the delete call still works for this use case.
The scope is then at provider scope, for example `/providers/microsoft.capacity`, and empty for assignemnts on root.

[the remark this answers is not in the document]
Comment (stephybun) [reply internal/services/authorization/parse/role_assignment.go:195]: Thanks for the confirmation @Felix-Franz! Would you mind fixing the typo in the error message? Once that's done we can merge this.

[the remark this answers is not in the document]
Comment (Felix-Franz) [reply internal/services/authorization/parse/role_assignment.go]: Oh sorry, I missed that somehow.
The fix is already commited.

**Review by stephybun — APPROVED** [review]
Thanks @Felix-Franz LGTM 🏖️

**Pull request merged**